crates/bevy_ecs/src/storage/sorted_small_vec.rs

```diff
@@ -0,0 +1,259 @@
+use smallvec::SmallVec;
+use core::cmp::Ordering;
+
+/// Stores a sorted list of indices with quick implementation for union, difference, intersection.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct SortedSmallVec<const N: usize>(SmallVec<[usize; N]>);
```

Comment:
I feel like the name here isn't clear. Maybe just `SmallVecSet`? `SortedSmallVec` doesn't really make it clear that it's a set, and the fact that it's sorted seems like an implementation detail (although this could be made as a guarantee on `ones`).

Reply:
+1 for the `Set` suffix.  That better matches names like `HashSet` and `BTreeSet`.  I'd vote for keeping `Sorted` in the name and making it part of the contract, since `Hash` and `BTree` are also implementation details.  But `SortedSmallVecSet` would be awfully verbose.

Reply:
I'll go for `SortedVecSet`